### ISSUE TITLE
Clippy fails when run with all features enabled

### ISSUE THREAD

**Issue by sbrocket** [serenity-rs/serenity#1423]
I discovered this in #1422, because shifting the workspace that was previously under examples/ up to the repo root means that the `cargo clippy --workspace --tests` in the Lint action now runs clippy against serenity with all features enabled, rather than just default features.

Without that change, you can run `cargo clippy --all-features --tests` and see that it currently fails.

To unblock #1422 I'm going to update lint.yml to remove the `--workspace` arg, so that clippy is checking the same features as before. Readding it will require fixing all the outstanding clippy lints. I'm going to put up another PR will all the automatic fixes clippy was able to make, but there are other manual fixes that need to be made like missing doc sections.
